src/service/navigation.js

```diff
@@ -529,24 +530,72 @@ export class Navigation {
-   * @param {!Location} targetLocation
-   * @param {!Location} currentLocation
+   * @param {!Location} toLocation
+   * @param {!Location} fromLocation
    * @private
    */
-  handleNavigation_(e, element, targetLocation, currentLocation) {
-    const target = getHrefMinusHash(targetLocation);
-    const current = getHrefMinusHash(currentLocation);
+  handleNavigation_(e, element, toLocation, fromLocation) {
+    const to = getHrefMinusHash(toLocation);
+    const from = getHrefMinusHash(fromLocation);
 
     // Handle same-page (hash) navigation separately.
-    if (targetLocation.hash && target == current) {
-      this.handleHashNavigation_(e, targetLocation, currentLocation);
+    if (toLocation.hash && to == from) {
+      this.handleHashNavigation_(e, toLocation, fromLocation);
     } else {
       // Otherwise, this is an other-page (external) navigation.
+      let target = (element.getAttribute('target') || '').toLowerCase();
+
       if (this.isEmbed_ || this.isInABox_) {
         // Target in the embed must be either _top or _blank (default).
-        const targetAttr = (element.getAttribute('target') || '').toLowerCase();
-        if (targetAttr != '_top' && targetAttr != '_blank') {
-          element.setAttribute('target', '_blank');
+        if (target != '_top' && target != '_blank') {
+          target = '_blank';
+          element.setAttribute('target', target);
+        }
+      }
+
+      // Safari 13: Temporarily remove viewer query params from iframe
+      // (e.g. amp_js_v, usqp) to prevent document.referrer from being reduced
+      // to eTLD+1 (e.g. ampproject.org).
+      const {win} = this.ampdoc;
+      const platform = Services.platformFor(win);
+      const viewer = Services.viewerForDoc(element);
+      if (
+        isExperimentOn(
+          this.ampdoc.win,
+          'remove-viewer-query-params-on-navigate'
+        ) &&
+        location.search &&
+        platform.isSafari() &&
+        platform.getMajorVersion() >= 13 &&
+        viewer.isProxyOrigin() &&
+        viewer.isEmbedded()
+      ) {
+        dev().info(
+          TAG,
+          'Removing query string before navigation:',
+          location.search
+        );
+        const original = location.href;
+        const noQuery = `${location.origin}${location.pathname}${location.hash}`;
+        win.history.replaceState(null, '', noQuery);
+
+        const restoreQuery = () => {
+          dev().info(TAG, 'Restored URL with query string:', original);
+          win.history.replaceState(null, '', original);
+        };
+
+        // For blank_, restore query params after the new page opens.
+        if (target === '_blank') {
+          win.setTimeout(restoreQuery, 0);
+        } else {
+          // For _top etc., wait until page is restored from page cache (bfcache).
+          // https://webkit.org/blog/516/webkit-page-cache-ii-the-unload-event/
+          win.addEventListener('pageshow', function onPageShow(e) {
```

Comment:
Does this prevent bfcache like unload does?

Reply:
Nope.